src/stable/components/visualHFSM/generate.cpp

```diff
@@ -478,3 +658,424 @@ std::string Generate::getCppName () {
+			}
+			this->fs.flush();
+			
+		}
+		if (id != 1) {
+
+			firstState = true;
+			for ( std::list<Node>::iterator nodeListIterator = nodeList.begin();
+					nodeListIterator != nodeList.end(); nodeListIterator++ ) {
+				
+				if (mapNameTime.find(nodeListIterator->getName()) != mapNameTime.end()) {
+					if (firstState){
+						this->fs << this->mapTab[T_THREE] << "else:" << std::endl;
+						this->fs << this->mapTab[T_FOUR];
+						this->fs << "if(self.sub" << id << " == \"" << nodeListIterator->getName() << "\"):" << std::endl;
+						firstState = false;
+					}else{
+						this->fs << this->mapTab[T_FOUR];
+						this->fs << "elif(self.sub" << id << " == \"" << nodeListIterator->getName() << "\"):" << std::endl;
+					}
+					this->fs << this->mapTab[T_FIVE];
+					this->fs << "t_" << nodeListIterator->getName() << "_max = " << mapNameTime[nodeListIterator->getName()] << " - (t_fin - t_ini)" << std::endl;
+					this->fs << this->mapTab[T_FIVE];
+					this->fs << "ghostStateIndex = self.StateSub" << id << ".index(self.sub" << id << ") + 1" << std::endl;					
+					this->fs << this->mapTab[T_FIVE];
+					this->fs << "sub" << id << " = self.StatesSub" << id << "[ghostStateIndex]" << std::endl;
+				} 
+			}
+		}
+		this->fs << std::endl;
+
+		this->fs << this->mapTab[T_THREE] << "totalb = time.time() * 1000000" << std::endl;
+		this->fs << this->mapTab[T_THREE] << "msecs = (totalb - totala) / 1000;" << std::endl;
+		this->fs << this->mapTab[T_THREE] << "if(msecs < 0 or msecs > cycle):" << std::endl;
+		this->fs << this->mapTab[T_FOUR] << "msecs = cycle" << std::endl;
+		this->fs << this->mapTab[T_THREE] << "else:" << std::endl;
+		this->fs << this->mapTab[T_FOUR] << "msecs = cycle - msecs" << std::endl;
+		this->fs << std::endl;
+		this->fs << this->mapTab[T_THREE] << "time.sleep(msecs / 1000)" << std::endl;
+		this->fs << this->mapTab[T_THREE] << "if(msecs < 33 ):" << std::endl;
+		this->fs << this->mapTab[T_FOUR] << "time.sleep(33 / 1000);" << std::endl;
+
+		this->fs << std::endl;
+		this->fs << std::endl;
+		this->fs.flush();
+	}
+}
+
+void Generate::generateConnectToProxys_py(){
+	this->fs << 
+"	def connectToProxys(self):\n\
+		self.ic = Ice.initialize(sys.argv)\n\
+\n";
+	boost::format fmt_iConnect( /* %1%: getName() %2%: getInterface() */
+"		# Contact to %1%\n\
+		%1% = self.ic.propertyToProxy('automata.%2%.Proxy')\n\
+		if(not %1%):\n\
+			raise Exception('could not create proxy with %1%')\n\
+		self.%1%Prx = %2%Prx.checkedCast(%1%)\n\
+		if(not self.%1%Prx):\n\
+			raise Exception('invalid proxy automata.%2%.Proxy')\n\
+		print '%1% connected'\n\
+\n");
+	for ( std::list<IceInterface>::iterator listInterfacesIterator = this->listInterfaces->begin();
+			listInterfacesIterator != this->listInterfaces->end(); listInterfacesIterator++ ) {
+		std::string iName = listInterfacesIterator->getName();
+		std::string iInterface = listInterfacesIterator->getInterface();
+		std::cout<<boost::str(fmt_iConnect % iName % iInterface);

```

Comment:
This line should be:

this->fs << boost::str(fmt_iConnect % iName % iInterface);

Not to std::cout